src/auth.rs

```diff
@@ -2,20 +2,23 @@ use rocket::State;
 use rocket_contrib::json::{Json, JsonValue};
 use diesel::prelude::*;
 use super::schema::users::dsl::*;
+use crate::db::PgPool;
 
-#[derive(Serialize, Deserialize)]
+#[derive(Queryable, Serialize, Deserialize)]
 pub struct Credentials {
+    id: i32,
     username: String,
-    password: String,
+    passwordhash: Option<String>,
+    emailaddress: Option<String>,
 }
 
-
 #[post("/auth/login", format = "json", data = "<credentials>")]
-pub fn login(pool: State<Pool>, credentials: Json<Credentials>) -> JsonValue {
-    let products_request = users.load(&*pool).expect(" blah"); 
+pub fn login(pool: State<PgPool>, credentials: Json<Credentials>) -> JsonValue {
+    let connection = pool.get().unwrap();
+    let products_request: Vec<Credentials> = users.load(&connection).expect(" blah");
```

Comment:
We have to pull out a connection from the pool using `pool.get().unwrap()`. Also, we need to explicitly use `Vec<Credentials>` so that it knows what type to store the resulting data in.